vendor/views/mixin.py

```diff
@@ -1,16 +1,39 @@
 from django.core.exceptions import ImproperlyConfigured
 from django.contrib import messages
-from django.utils.translation import ugettext as _
-from django.http import Http404
 from django.contrib.sites.models import Site
+from django.contrib.sites.shortcuts import get_current_site
+from django.http import Http404
 from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
 
 
-class ProductRequiredMixin():
+def get_site_from_request(request):
+    if hasattr(request, 'site'):
+        return request.site
+    return get_current_site(request)
+
+
+class SetSiteToRequestMixin:
+
+    def dispatch(self, request, *args, **kwargs):
+        request.site = get_site_from_request(request)
```

Comment:
This is not the right way to do this. Modifying a request in a view is considered a no-no.  see notes below.  It is possible a middleware uses it on the backside of a response.

Reply:
The point of creating the function is to use it wherever it is needed.

Reply:
Got it, thought it would be much cleaner just to add it to the request in the dispatch. I have removed the mixin and now using only using the function instead. :)